### PR TITLE
e2e: Add chatqna workload initial details

### PR DESCRIPTION
**Pull request by vbedida79**
Pr includes chatqna workload setup's initial details in a readme and the corresponding tgi-gaudi serving runtime yaml

Signed-off-by: vbedida79 <veenadhari.bedida@intel.com>

### CODE REVIEW

File: e2e/chatqna/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# OPEA ChatQnA workload on OCP
+
```
Comment (uMartinXu) [inline e2e/chatqna/README.md:2]: According to the architecture diagram, OPEA is the reference workload for the AI foundation, so suggest to put it into folder  workloads/OPEA/chatqna. @chaitanya1731 @hershpa  any comments?

Comment (hershpa) [reply e2e/chatqna/README.md:2]: That sounds reasonable. A new workloads directory for reference workloads, some of which will be from OPEA.

File: e2e/chatqna/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+### ChatQnA backend- langchain and redis vector database
+* Please follow the [docker-compose file](https://github.com/opea-project/GenAIExamples/blob/main/ChatQnA/deprecated/langchain/docker/docker-compose.yml). To create the backend image, use the [dockerfile](https://github.com/opea-project/GenAIExamples/blob/main/ChatQnA/deprecated/langchain/docker/Dockerfile). Make sure to mount the langchain folder while building image and deploying the pod. The redis vector database image used is ``` redis/redis-stack:7.2.0-v9 ```. Please update the environment variables according to your setup and model variables in the yaml's.
+
+### Minio storage for models
+Minio/S3 storage format is supported on RHOAI. To setup minio on OCP, please follow the steps in [documentation](https://ai-on-openshift.io/tools-and-applications/minio/minio/#create-a-matching-data-connection-for-minio). Make sure to allocate enough storage based on the model weights. Once the minio-ui can be accessed, create the bucket and add your model files. Also create an access key and secret key.
```
Comment (uMartinXu) [inline e2e/chatqna/README.md:20]: Suggest to put "Storage Setup" into prerequisites.

File: e2e/chatqna/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+## Prerequisites
+* Provisioned RHOCP cluster. Follow steps [here](/README.md#provisioning-rhocp-cluster)
+* Provisioning Intel Gaudi 2 accelerator:
+  * Setup KMM and node feature discovery (NFD) for drivers. Follow the steps [here](https://docs.habana.ai/en/latest/Orchestration/HabanaAI_Operator/Environment_Setup.html)
+  * Setup HabanaAI operator and create device config custom Resource. Follow the steps [here](https://docs.habana.ai/en/latest/Orchestration/HabanaAI_Operator/Deploying_HabanaAI_Operator.html)
+* Install RHOAI. Follow steps [here](../inference/README.md/#install-rhoai)
+
+## Deploy workload
+* The workload uses docker-compose yaml's. They can be converted into OCP compatible yaml's using [Kompose tool](https://kubernetes.io/docs/tasks/configure-pod-container/translate-compose-kubernetes/). This creates the corresponding deployment and service yaml's.
```
Comment (uMartinXu) [inline e2e/chatqna/README.md:14]: Do we really need this?

File: e2e/chatqna/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+* Please follow the [docker-compose file](https://github.com/opea-project/GenAIExamples/blob/main/ChatQnA/deprecated/langchain/docker/docker-compose.yml). To create the backend image, use the [dockerfile](https://github.com/opea-project/GenAIExamples/blob/main/ChatQnA/deprecated/langchain/docker/Dockerfile). Make sure to mount the langchain folder while building image and deploying the pod. The redis vector database image used is ``` redis/redis-stack:7.2.0-v9 ```. Please update the environment variables according to your setup and model variables in the yaml's.
+
+### Minio storage for models
+Minio/S3 storage format is supported on RHOAI. To setup minio on OCP, please follow the steps in [documentation](https://ai-on-openshift.io/tools-and-applications/minio/minio/#create-a-matching-data-connection-for-minio). Make sure to allocate enough storage based on the model weights. Once the minio-ui can be accessed, create the bucket and add your model files. Also create an access key and secret key.
+
+### Text Generation Inference for Gaudi on RHOAI
+1. Access rhods-dashboard link from routes on your web console and click on Data Science Projects. Now, create a project.
```
Comment (uMartinXu) [inline e2e/chatqna/README.md:23]: Can we have a few snapshots for this?

File: e2e/chatqna/README.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+2. Click on the project, and choose ```data connection``` section. In the fields, add your access and secret keys from minio, along with minio-api route (from your minio deployment in previous step). This sets up the connection for your model to be downloaded and prepare for serving.
```
Comment (uMartinXu) [inline e2e/chatqna/README.md:24]: Do we have to do it manually? Do we have a yaml file to do it automatically?

File: e2e/chatqna/README.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+3. Now click on the Settings and choose ```ServingRuntime```. Copy or import the ```tgi-gaudi-servingruntime.yaml```. The [tgi-gaudi](https://github.com/huggingface/tgi-gaudi) serving runtime is used.
```
Comment (uMartinXu) [inline e2e/chatqna/README.md:25]: can with do it with the help of yaml file not do it manually?

Comment (uMartinXu) [reply e2e/chatqna/README.md:25]: do we have instructions for testing the application? Or it is in the following PRs?

File: e2e/chatqna/README.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
+# OPEA ChatQnA workload on OCP
+
+## Overview
+This workload is based on the [OPEA ChatQnA](https://github.com/opea-project/GenAIExamples/tree/main/ChatQnA/deprecated) with [Intel® Gaudi 2 accelerator](https://habana.ai/products/gaudi2/) and [Red Hat OpenShift AI](https://www.redhat.com/en/technologies/cloud-computing/openshift/openshift-ai) (RHOAI) on RHOCP. It uses RAG with Redis vector-db and text generation inference server on RHOAI and Intel Gaudi2 accelerator. Refer to the OPEA readme for more details about the [workload architecture](https://github.com/opea-project/GenAIExamples/tree/main/ChatQnA#chatqna-application). 
+
+## Prerequisites
+* Provisioned RHOCP cluster. Follow steps [here](/README.md#provisioning-rhocp-cluster)
+* Provisioning Intel Gaudi 2 accelerator:
```
[the remark this answers is not in the document]
Comment (vbedida79) [reply e2e/chatqna/README.md:8]: Will update the links after that pr is merged

File: e2e/chatqna/README.md (hunk 3 above)
Comment (vbedida79) [reply e2e/chatqna/README.md:14]: docker-compose files are not directly compatible on k8s/ocp. this tool converts to k8s worklods (deploymentconfig and service)

File: e2e/chatqna/README.md (hunk 5 above)
Comment (vbedida79) [reply e2e/chatqna/README.md:24]: This is on the web console of rhoai, we have to add it there. no yaml's

File: e2e/chatqna/README.md (hunk 6 above)
Comment (vbedida79) [reply e2e/chatqna/README.md:25]: > can with do it with the help of yaml file not do it manually?

serving runtime has to be created on rhoai web console, cannot be created with oc apply

Comment (vbedida79) [reply e2e/chatqna/README.md:25]: > do we have instructions for testing the application? Or it is in the following PRs?

no testing instructions as of now

File: e2e/chatqna/README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (chaitanya1731) [reply e2e/chatqna/README.md:8]: Yes.. We can update the link to the new gaudi README.

File: e2e/chatqna/README.md (hunk 1 above)
Comment (chaitanya1731) [reply e2e/chatqna/README.md:2]: Yes. that sounds like a good idea.

**Comment by vbedida79** [comment]
Closing PR, submitting new one.

**Pull request closed without merging**